frontend/src/components/pages/RegistrantExperience/index.tsx

```diff
@@ -109,62 +166,169 @@ const RegistrantExperiencePage = (): React.ReactElement => {
+
+          setRegisteringCampers(restoredSession.campers);
+          setCamp(restoredSession.camp);
+          waiverDispatch({
+            type: WaiverActions.LOADED_WAIVER,
+            waiver: restoredSession.waiver,
+          });
+          setCheckoutUrl(restoredSession.checkoutUrl);
+
+          if (registrationResult === CANCEL_RESULT_CODE) {
+            // TODO need the isFilled booleans to point to fields on campers
+            // to fill out status bar properly
+            setSamplePersonalInfo(true);
+            setSampleAdditionalInfo(true);
+            setIsLoading(false);
+            setCurrentStep(RegistrantExperienceSteps.ReviewRegistrationPage);
+
+            setErrorModalMessage({
+              title: "Payment cancelled",
+              body:
+                "No payment was processed. If this was not intentional, please try again.",
+            });
+            errorModalOnOpen();
+          }
+
+          sessionWasRestored = true;
+        } catch (err: unknown) {
+          // eslint-disable-line no-empty
+        }
+      }
+    }
+
+    // If we expect session cache to exist, and it doesn't exist, do not repopulate
+    // via network call -- instead show generic message in RegistrationResult in case
+    // the user entered `?result=success` manually into the URL (so no actually success)
+    if (!sessionWasRestored && registrationResult !== SUCCESS_RESULT_CODE) {
+      CampsAPIClient.getCampById(campId).then((campResponse) => {
+        if (campResponse.id) {
+          setCamp(campResponse);
+        }
+        setIsLoading(false);
+      });
+
+      AdminAPIClient.getWaiver().then((waiver) => {
+        if (waiver.clauses) {
+          waiverDispatch({
+            type: WaiverActions.LOADED_WAIVER,
+            waiver,
+          });
+        }
+
+        // TODO error handling
+      });
+    }
+  }, [campId, errorModalOnOpen, registrationResult]);
+
   return (
-    <Flex
-      direction="column"
-      w="100vw"
-      pt={{ sm: "120px", md: "120px", lg: "144px" }}
-      pb={{ sm: "170px", md: "108px", lg: "144px" }}
-      justifyContent="flex-start"
-    >
-      <RegistrationNavStepper
-        currentStep={currentStep}
-        isPersonalInfoFilled={isPersonalInfoFilled}
-        isAdditionalInfoFilled={isAdditionalInfoFilled}
-        isWaiverFilled={isWaiverFilled}
-        isReviewRegistrationFilled={isReviewRegistrationFilled}
-        setCurrentStep={setCurrentStep}
-      />
-      {!isLoading && (
-        <Box mx="10vw">
-          {camp ? (
-            getCurrentRegistrantStepComponent(currentStep)
-          ) : (
-            <Text>Error: Camp not found. Please go back and try again.</Text>
+    <>
+      {registrationResult === SUCCESS_RESULT_CODE ? (
+        <RegistrationResultPage
+          camp={camp}
+          campers={registeringCampers}
+          items={camp ? mapToCampItems(camp, registeringCampers) : undefined}
```

Comment:
can the camp ever be undefined here? If so, would it make more sense to check for that in the top condition so we don't render the RegistrationResultPage until we have the camp data?

Reply:
should be resolved by session selection, ie. never go the registration steps without a camp